src/templates/program-page.js

```diff
@@ -259,224 +216,170 @@ function renderProgramInfo (courseData, courseNotes, variantDataHeading, variant
 
-                                        // After:
-                                        // const Named = () => {}
-                                        // export default Named;
-
-
-    const ProgramPage = ({data, location}) => {
-    let callToActionData = [];
-    let careerData;
-    let courseData;
-    let degreesData;
-    let domains;
-    let employerData;
-    let footerData;
-    let imageData = [];
-    let imageTaggedData = [];
-    let progData;
-    let newsData;
-    let specData;
-    let statsData;
-    let tagData;
-    let testimonialData;
-    let variantData;
-    let videoData;
-
-    // set data
-    if (data.careers.edges[0] !== undefined) { careerData = data.careers.edges; }
-    if (data.ctas.edges[0] !== undefined) { callToActionData = data.ctas.edges; }
-    if (data.employers.edges[0] !== undefined) { employerData = data.employers.edges; }
-    if (data.footer.edges[0] !== undefined) { footerData = data.footer.edges; }
-    if (data.news.edges[0] !== undefined) { newsData = data.news.edges; }
-    if (data.programs.edges[0] !== undefined) { progData = data.programs.edges[0].node; }
-    if (progData.field_domain_access !== undefined) { domains = progData.field_domain_access; }
-    if (progData.relationships.field_courses !== undefined) { courseData = progData.relationships.field_courses; }
-    if (progData.relationships.field_program_statistics !== undefined) { statsData = progData.relationships.field_program_statistics; }
-    if (data.testimonials.edges[0] !== undefined) { testimonialData = data.testimonials.edges; }
-    if (data.images.edges !== undefined) { imageData = data.images.edges; }
-    if (data.imagesTagged.edges !== undefined) { imageTaggedData = data.imagesTagged.edges; }
-    if (data.videos.edges[0] !== undefined) { videoData = data.videos.edges[0].node; }
-    
+const ProgramPage = ({data, location}) => {
+        
+    let progData = data.programs.edges[0]?.node;
+    let callToActionData = data.ctas?.edges;
+    let careerData = data.careers?.edges;
+    let courseData = progData.relationships?.field_courses;
+    let degreesData = progData.relationships?.field_degrees;
+    let domains = progData?.field_domain_access;
+    let employerData = data.employers?.edges;
+    let footerData = data.footer?.edges;
+    let imageData =  data.images?.edges;
+    let imageTaggedData = data.imagesTagged?.edges;    
+    let newsData = data.news?.edges;
+    let specData = progData.relationships?.field_specializations;
+    let statsData = progData.relationships?.field_program_statistics;
+    let testimonialData = data.testimonials?.edges;
+    let variantData = progData.relationships?.field_program_variants;
+    let variantDataHeading = prepareVariantHeading(variantData);
+    let videoData = data.videos.edges[0]?.node;
+  
     const heroImage = (contentExists(imageData) ? imageData : (contentExists(imageTaggedData) ? imageTaggedData : null));
```

Comment:
Lately, I've been wondering if we should just drop the contentExists function completely.

```suggestion
    const heroImage = (imageData?.length>0 ? imageData : (imageTaggedData?.length>0 ? imageTaggedData : null));
```